src/Todo/Todo.js

```diff
@@ -0,0 +1,100 @@
+import React from "react";
+import { useState } from "react";
+
+export const Todo = () => {
+    const [input, setInput] = useState({
+        item: "",
+        isComplete: false
+    })
+    const [showItem, setShowItem] = useState([])
+    const [error, setError] = useState(false)
+    const [edit, setEdit] = useState(-1)
+    const handleChange = (e) => {
+        const { name, value } = e.target
+        setInput((data) => {
+            return { ...data, [name]: value }
+        })
+        setError(false)
+    }
+
+    const submit = (e) => {
+        e.preventDefault()
+        if (input.item.trim() === "") {
+            setError("Please enter a text.");
+            return;
+        }
+        if (edit >= 0) {
+            setShowItem((prevoiusItem) => {
+                const updatedItems = prevoiusItem.map((item, index) => {
+                    if (index === edit) {
+                        return { ...item, item: input.item };
+                    }
+                    return item;
+                });
+                return updatedItems;
+            });
+            setInput({ item: "" });
+            setEdit(-1);
+        } else {
+            setShowItem([...showItem, input]);
+            setInput({ item: "" });
+        }
+        setError(false);
+    }
+
+    const handleCheckBox = (currentIndex) => {
+        let data = [...showItem]
+        data[currentIndex].isComplete = !data[currentIndex].isComplete
+        setShowItem([...data])
+    }
+
+    const handleEditChange = (index) => {
```

Comment:
why passing the index? why not full item? so you do not need to do the find @amrit56789